waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -49,6 +50,7 @@ type
     ethContractAddress*: string
     ethRpc*: Option[Web3]
```

Comment:
not strictly related to the PR but related to my other comments.

any reason for having `ethRpc` as an option? I believe every onchain group manager would need one, or?

the less options the less we have to check if `isSome` and the less risk we have of doing `.get` in a `none`.

Reply:
The reason the ethRpc is an option is because it's set in the `init*` method. all methods which utilize the optional fields, like `latestProcessedBlock` and `rlnContractDeployedBlockNumber`, are guarded by https://github.com/waku-org/nwaku/blob/3674879b0e1c072303b9d6524462c20656c9a82b/waku/waku_rln_relay/group_manager/on_chain/group_manager.nim#L73-L75 which ensures that the groupManager is initialized before any other operations are performed on it (and by extension, the `ethRpc`, `latestProcessedBlock`, and `rlnContractDeployedBlockNumber`)
I could change them to be non-options, but they'd be nil for a while, and if someone accessed them before init, then would throw a segfault, right?

Reply:
ah right, feel free to leave the `ethRpc` as it is.

it was a personal preference. i prefer to check for `isNil` which then avoids `.get()`, its cleaner due to `ethRpc*: Web3` and easier to read (no templates). biased by golang i guess :)